src/main/kotlin/org/opensearch/replication/ReplicationTranslogDeletionPolicy.kt

```diff
@@ -0,0 +1,54 @@
+package org.opensearch.replication
+
+import org.opensearch.index.seqno.RetentionLease
+import org.opensearch.index.seqno.RetentionLeases
+import org.opensearch.index.translog.TranslogDeletionPolicy
+import org.opensearch.index.translog.TranslogReader
+import org.opensearch.index.translog.TranslogWriter
+import java.io.IOException
+import java.util.function.Supplier
+
+class ReplicationTranslogDeletionPolicy(
+    private val retentionSizeInBytes: Long, retentionAgeInMillis: Long,
+    private val retentionTotalFiles: Int,
+    private val retentionLeasesSupplier: Supplier<RetentionLeases>
+) : TranslogDeletionPolicy(
+    retentionSizeInBytes, retentionAgeInMillis, retentionTotalFiles
+) {
+
+    /**
+     * returns the minimum translog generation that is still required by the system. Any generation below
+     * the returned value may be safely deleted
+     *
+     * @param readers current translog readers
+     * @param writer  current translog writer
+     */
+    @Synchronized
+    @Throws(IOException::class)
+    override fun minTranslogGenRequired(readers: List<TranslogReader>, writer: TranslogWriter): Long {
+        val minBySize: Long = getMinTranslogGenBySize(readers, writer, retentionSizeInBytes)
+        val minByRetentionLeases: Long = getMinTranslogGenByRetentionLease(readers, writer)
+        val minByTranslogGenSettings = super.minTranslogGenRequired(readers, writer)
```

Comment:
In this path `getMinTranslogGenBySize` will be invoked twice, once here and once by `super. minTranslogGenRequired` if this is too expensive, can revert back to updating the entire method.